recommenders/utils/spark_utils.py

```diff
@@ -11,37 +11,50 @@
     pass  # skip this import if we are in pure python environment
 
 
+MMLSPARK_PACKAGE = "com.microsoft.ml.spark:mmlspark_2.11:0.18.1"
+MMLSPARK_REPO = "https://mvnrepository.com/artifact"
+
 def start_or_get_spark(
     app_name="Sample",
     url="local[*]",
     memory="10g",
     config=None,
     packages=None,
     jars=None,
-    repository=None,
+    repositories=None,
+    mmlspark_package=None,
+    mmlspark_repository=None
```

Comment:
I saw the comment of @gramhagen and @yueguoguo here: https://github.com/microsoft/recommenders/issues/1439 about adding mmlspark in the signature.

Another approach could be the following, having a generic function that could input any package, jar, etc:
```
def start_or_get_spark(
    app_name="Sample",
    url="local[*]",
    memory="10g",
    config=None,
    packages=None,
    jars=None,
repositories=None
)
```
and then a second function that includes specifically mmlspark, something like:
```
def start_or_get_spark_with_mmlspark(
    app_name="Sample",
    url="local[*]",
    memory="10g",
    config=None,
    packages=None,
    jars=None,
repositories=None

):
packages_full = mmlspark_package
if packages is not None:
    package_full+="".format(",".join(new_packages))
# same with repositories
```
The advantage of this is that we have a generalistic function for spark that can get any package, and another one explicitely for mmlspark.

What do you think @yueguoguo , @gramhagen , @anargyri? Happy to back off if you all guys agree we should go the original approach

Reply:
🤔 I don't see a clear benefit from duplicating the function (the code of the two functions will be almost identical). 
I think the Python philosophy of specific rather than generalistic applies mainly when the functionalities differ significantly. In this case it's the same functionality essentially, just the repo info changes.

Reply:
is it not possible to add mmlspark using the current packages/repositories inputs? just wondering if it's worth hiding that from the user and creating a library specific option. my preference would be to let the user provide these values and we can provide examples.

Reply:
Yes, I was following your earlier suggestion, but this is another good option, to have just one function and let the user define the list of packages and repositories (If I understand correctly what you suggest).

Reply:
right, would be good to test if this works
```
start_or_get_spark(packages=["com.microsoft.ml.spark:mmlspark_2.11:0.18.1"], repositories=["https://mvnrepository.com/artifact"])
```

Reply:
This was already working [here](https://github.com/microsoft/recommenders/blob/main/examples/02_model_content_based_filtering/mmlspark_lightgbm_criteo.ipynb) (only change was the repos is now a list).

Reply:
got it makes sense, if you guys think that two functions (one general, one with mmlspark) are unnecessary, then I would lean more towards having just one function where we can have any arbitrary input:
`start_or_get_spark(packages=["com.microsoft.ml.spark:mmlspark_2.11:0.18.1"], repositories=["https://mvnrepository.com/artifact"])`, rather than a function with the specific mmlspark inputs:
```
start_or_get_spark(
    app_name="Sample",
    url="local[*]",
    memory="10g",
    config=None,
    packages=None,
    jars=None,
    repository=None,
    repositories=None,
    mmlspark_package=None,
    mmlspark_repository=None
```

Reply:
Ok, I will keep a single, simpler version then.

Reply:
@miguelgfierro @gramhagen if the recent changes look good, could you approve of the PR?